### PR TITLE
Include prettier in npm test run

### PR DESCRIPTION
**Pull request by leebyron**
Applied to all files

### CODE REVIEW

File: pages/lib/genTypeDefData.js [hunk 1]
```diff
@@ -189,15 +188,15 @@ function DocVisitor(source) {
       ensureGroup(node);
 
       var propertyObj = {
-        line: getLineNum(node),
+        line: getLineNum(node)
```
Comment (vjeux) [inline pages/lib/genTypeDefData.js]: FYI, you can add `--trailing-comma=es5` or `--trailing-comma=all` to add them back

File: pages/lib/markdown.js [hunk 2]
```diff
@@ -126,14 +141,18 @@ function findPath(context, search) {
...
+    if (
+      path.reduce(
+        (def, name) =>
+          def &&
+          ((def.module && def.module[name]) ||
+            (def.interface &&
+              allMembers &&
+              allMembers.get(def.interface)[name]) ||
+            undefined),
+        { module: defs }
```
Comment (vjeux) [inline pages/lib/markdown.js]: you can use `--no-bracket-spacing` to remove spaces here. It seems like at fb we don't add it there.

File: pages/lib/markdownDocs.js [hunk 3]
```diff
@@ -1,33 +1,26 @@
...
-                signatures: member.signatures }
-            )
-          )
-        );
+        Seq(typeDef.interface.groups).forEach(group => Seq(
+          group.members
+        ).forEach((member, memberName) => markdownDoc(member.doc, {
+          typePath: typePath.concat(memberName.slice(1)),
+          signatures: member.signatures
+        })));
```
Comment (vjeux) [inline pages/lib/markdownDocs.js]: This is not printed ideally. I'm sure we can get closer to the original one.

File: pages/src/docs/src/Defs.js [hunk 4]
```diff
@@ -172,7 +210,8 @@ var TypeDef = React.createClass({
       <span
         className={'t ' + className}
         onMouseOver={this.mouseOver}
-        onMouseOut={this.mouseOut}>
+        onMouseOut={this.mouseOut}
```
Comment (vjeux) [inline pages/src/docs/src/Defs.js]: you can use `--jsx-bracket-same-line` in order to use fb-style brackets

File: pages/src/docs/src/Defs.js [hunk 5]
```diff
@@ -200,54 +238,56 @@ var MemberDef = React.createClass({
...
-    p.name.length +
-    (p.optional ? 3 : 2) +
-    typeLength(info, p.type),
+  return params.reduce(
+    (s, p) =>
+      s +
+      (p.varArgs ? 3 : 0) +
+      p.name.length +
+      (p.optional ? 3 : 2) +
+      typeLength(info, p.type),
```
Comment (vjeux) [inline pages/src/docs/src/Defs.js]: the indentation is nice, before if was unclear at a glance that there was a second argument :)

File: pages/lib/markdownDocs.js (hunk 3 above)
Comment (vjeux) [inline pages/lib/markdownDocs.js]: https://github.com/prettier/prettier/issues/959

File: pages/src/docs/src/SideBar.js [hunk 6]
```diff
@@ -47,64 +54,77 @@ var SideBar = React.createClass({
...
-                    {member.memberName + (member.memberDef.signatures ? '()' : '')}
-                  </Router.Link>
-                </div>
+          <section>
+            {Seq(memberGroups)
+              .map(
+                (members, title) => members.length === 0
+                  ? null
+                  : Seq([
+                      (
```
Comment (vjeux) [inline pages/src/docs/src/SideBar.js]: the extra `(` is fixed in master

File: pages/src/docs/src/SideBar.js [hunk 7]
```diff
@@ -47,64 +54,77 @@ var SideBar = React.createClass({
...
+                      ),
+                      Seq(members).map(member => (
+                        <div key={member.memberName}>
+                          <Router.Link
+                            to={'/' + typeName + '/' + member.memberName}
+                          >
+                            {
+                              member.memberName +
+                                (member.memberDef.signatures ? '()' : '')
+                            }
```
Comment (vjeux) [inline pages/src/docs/src/SideBar.js]: https://github.com/prettier/prettier/issues/961 we should not indent this.

File: pages/src/docs/src/TypeDocumentation.js [hunk 8]
```diff
@@ -46,37 +49,34 @@ var TypeDocumentation = React.createClass({
...
+            focus={name}
+            memberGroups={memberGroups}
+            toggleShowInherited={this.toggleShowInherited}
+            toggleShowInGroups={this.toggleShowInGroups}
+            showInGroups={this.state.showInGroups}
+            showInherited={this.state.showInherited}
+          />}
         <div key={name} className="docContents">
 
-          {!def ?
```
Comment (vjeux) [inline pages/src/docs/src/TypeDocumentation.js]: this is an interesting way to format ternaries.

File: pages/src/src/Header.js [hunk 9]
```diff
@@ -57,68 +59,76 @@ var Header = React.createClass({
...
-              <a href="https://stackoverflow.com/questions/tagged/immutable.js?sort=votes">Questions</a>
-              <a href="https://github.com/facebook/immutable-js/">GitHub</a>
+          <div className="cover">
+            <div className="coverFixed">
+              <div className="filler">
+                <div className="miniHeaderContents">
+                  <a href="docs/" target="_self">Docs</a>
+                  <a
+                    href="https://stackoverflow.com/questions/tagged/immutable.js?sort=votes"
+                  >
```
Comment (vjeux) [inline pages/src/src/Header.js]: @rattrayalex didn't you made those not break even if they are > 80 columns?

File: src/List.js [hunk 10]
```diff
@@ -88,17 +104,17 @@ export class List extends IndexedCollection {
       return this;
     }
     if (this.__ownerID) {
-      this.size = this._origin = this._capacity = 0;
+      this.size = (this._origin = (this._capacity = 0));
```
Comment (vjeux) [inline src/List.js]: I just created https://github.com/prettier/prettier/issues/962 for it

File: src/List.js [hunk 11]
```diff
@@ -88,17 +104,17 @@ export class List extends IndexedCollection {
...
+      this._root = (this._tail = null);
       this.__hash = undefined;
       this.__altered = true;
       return this;
     }
     return emptyList();
   }
 
-  push(/*...values*/) {
+  push /*...values*/() {
```
Comment (vjeux) [inline src/List.js]: This will be fixed by https://github.com/prettier/prettier/pull/957

File: src/List.js [hunk 12]
```diff
@@ -269,15 +292,16 @@ class VNode {
     if (index === (level ? 1 << level : 0) || this.array.length === 0) {
       return this;
     }
-    var sizeIndex = ((index - 1) >>> level) & MASK;
+    var sizeIndex = index - 1 >>> level & MASK;
```
Comment (vjeux) [inline src/List.js]: Another data point for https://github.com/prettier/prettier/issues/872

File: pages/src/src/Header.js (hunk 9 above)
Comment (rattrayalex) [inline pages/src/src/Header.js]: Yeah, I definitely thought so... This definitely looks like it'd apply.

**Comment by vjeux** [comment]
Thanks for the real world usage. I filed a few issues on prettier repo for things we can improve!

**Pull request merged**